python-sdk/tests/files/locations/test_sftp.py

```diff
@@ -0,0 +1,80 @@
+import json
+from unittest.mock import patch
+
+import pytest
+from airflow.models.connection import Connection
+from airflow.providers.sftp.hooks.sftp import SFTPHook
+
+from astro.exceptions import PermissionNotSetError
+from astro.files.locations import create_file_location
+
+
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_connection")
+def test_get_transport_params_for_sftp(mock_sftp_hook):
+    """test get_transport_params() method which should return connect_kwargs when keyfile is passed"""
+    mock_sftp_hook.return_value = Connection(
+        conn_id="sftp_default",
+        conn_type="test",
+        login=1234,
+        host="localhost",
+        extra=json.dumps({"key_file": "/some/local/path/rsa.pem"}),
+    )
+    path = "sftp://bucket/some-file"
+    location = create_file_location(path)
+    credentials = location.transport_params
+    assert credentials == {"connect_kwargs": {"key_filename": "/some/local/path/rsa.pem"}}
+
+
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_connection")
+def test_get_transport_params_for_sftp_no_value(mock_sftp_hook):
+    """test get_transport_params() method when no keyfile is passed"""
+    mock_sftp_hook.return_value = Connection(
+        conn_id="sftp_default",
+        conn_type="test",
+        login=1234,
+        host="localhost",
+    )
+    path = "sftp://bucket/some-file"
+    location = create_file_location(path)
+    with pytest.raises(PermissionNotSetError):
+        location.transport_params
+
+
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_connection")
+def test_get_transport_params_for_sftp_password(mock_sftp_hook):
+    """test get_transport_params() method when no keyfile is passed"""
+    mock_sftp_hook.return_value = Connection(
+        conn_id="sftp_default", conn_type="test", login=1234, host="localhost", password="test"
+    )
+    path = "sftp://bucket/some-file"
+    location = create_file_location(path)
+    credentials = location.transport_params
+    assert credentials == {"connect_kwargs": {"password": "test"}}
+
+
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_tree_map")
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_connection")
+def test_get_paths_from_sftp(mock_sftp_conn, mock_list):
+    """Get the list of files from the sftp path"""
+    mock_sftp_conn.return_value = Connection(
+        uri="sftp://user@host:1234",
+    )
+    mock_list.return_value = (["some/sample.csv"],)
+    location = create_file_location("sftp://some/")
+    assert sorted(location.paths) == sorted(["sftp://user@host:1234/some/sample.csv"])
+
+
+@patch("airflow.providers.sftp.hooks.sftp.SFTPHook.get_conn")
+def test_size(mock_get_conn):
+    """Test get_size() of for SFTP file."""
```

Comment:
I see that the size also returns -1. Should we test for that as well?

Reply:
made it paramterized for both cases